### PR TITLE
#6 two factor authentication  &  Student CRUD

### PR DESCRIPTION
**Pull request by itsgracian**
#### Introduction
Authenticate using two-factor authentication && CRUD of Student

#### Description of Task
- I have created the subscription modules
- I have created subscription service, helper and middlewares
- I have created student modules
- I created auth middlewares for subscribers

#### Test PR
- create subscription - send POST on **[localhost:5000/api/v1/subscriptions/create-subscription](localhost:5000/api/v1/subscriptions/create-subscription)**
`{
	"phoneNumber":"",
	"name":""
}`
after user will receive SMS code
- authenticate - POST on **[localhost:5000/api/v1/subscriptions/authenticate-subscriber](localhost:5000/api/v1/subscriptions/authenticate-subscriber)**
`{
	"phoneNumber":"",
	"code":""
}`
- user will receive token or auth error
- to login as subscriber -> POST **[localhost:5000/api/v1/subscriptions/login](localhost:5000/api/v1/subscriptions/login)**
`{
	"phoneNumber":""
}` after checking phone number, user will receive SMS verification code on his mobile phone 
- view my profile as subscriber - GET **[localhost:5000/api/v1/subscriptions/read-profile](localhost:5000/api/v1/subscriptions/read-profile)**
- add student as a parent **[localhost:5000/api/v1/students/create-student](localhost:5000/api/v1/students/create-student)**
`{
	"name":"kwitonda",
	"school":"gitisi",
	"classStudy":"1"
}`
- update student as a parent **[localhost:5000/api/v1/students/update-student/3](localhost:5000/api/v1/students/update-student/3)**
`{
	"name":"kwitonda jean paul",
	"school":"gitisi",
	"classStudy":"1"
}`
- delete student as a parent **[localhost:5000/api/v1/students/delete-student/3](localhost:5000/api/v1/students/delete-student/3)**

N.B: student actions should be acted by the authenticated parent
#### Author
[gratien](https://github.com/itsgracian)

#### Story ID
[#6](https://trello.com/c/XDwjOlDh/6-subscription-with-sms-auth)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -2,4 +2,8 @@ DB_NAME=
 DB_PASSWORD= 
 DB_USER=
 DB_HOST=
-SECRET_OR_KEY = 
\ No newline at end of file
+SECRET_OR_KEY = 
+INTOUCH_USERNAME=
+INTOUCH_PASSWORD=
+INTOUCH_API_URL=
```
Comment (jnkindi) [inline .env.example:8]: Kindly remove these env variables as they are not used.

File: package.json [hunk 2]
```diff
@@ -29,6 +29,8 @@
     "@babel/polyfill": "^7.8.7",
     "@babel/preset-env": "^7.9.6",
     "@babel/register": "^7.9.0",
+    "africastalking": "^0.4.5-beta.0",
```
Comment (jnkindi) [inline package.json:32]: I think this package has not been used yet. You can remove it for now.

File: .env.example (hunk 1 above)
Comment (itsgracian) [reply .env.example]: @jnkindi removed. thanks

File: package.json (hunk 2 above)
Comment (itsgracian) [reply package.json:32]: @jnkindi  because we will use it, I think it would be better if it stays.

**Comment by itsgracian** [comment]
> Nice job @itsgracian however, is it necessary to add subject when creating a student? I think the class is the only thing we need about academics.

Hey @ericnyirimana  thanks, that's a great idea. let me change it.

**Pull request merged**